packages/expo-cli/src/credentials/local/credentialsJson.ts

```diff
@@ -0,0 +1,114 @@
+import path from 'path';
+
+import Joi from '@hapi/joi';
+import fs from 'fs-extra';
+
+import { Keystore } from '../credentials';
+
+interface CredentialsJson {
+  android?: {
+    keystore: {
+      keystorePath: string;
+      keystorePassword: string;
+      keyAlias: string;
+      keyPassword: string;
+    };
+  };
+  ios?: {
+    provisioningProfilePath: string;
+    distributionCertificate: {
+      path: string;
+      password: string;
+    };
+  };
+}
+
+const CredentialsJsonSchema = Joi.object({
+  android: Joi.object({
+    keystore: Joi.object({
+      keystorePath: Joi.string().required(),
+      keystorePassword: Joi.string().required(),
+      keyAlias: Joi.string().required(),
+      keyPassword: Joi.string().required(),
+    }),
+  }),
+  ios: Joi.object({
+    provisioningProfilePath: Joi.string().required(),
+    distributionCertificate: Joi.object({
+      path: Joi.string().required(),
+      password: Joi.string().required(),
+    }).required(),
+  }),
+});
+
+interface AndroidCredentials {
+  keystore: Keystore;
+}
+
+interface iOSCredentials {
+  provisioningProfile: string;
+  distributionCertificate: {
+    certP12: string;
+    certPassword: string;
+  };
+}
+
+async function exists(projectDir: string): Promise<boolean> {
```

Comment:
```suggestion
async function credentialsJsonExistsAsync(projectDir: string): Promise<boolean> {
```

Reply:
Please add `...Async` prefix to other async functions.

Reply:
it would be called with `credentialsJson.credentialsJsonExistsAsync`. I'm switching the name to `fileExistsAsync`.